src/jit/morph.cpp

```diff
@@ -12791,6 +12791,12 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
 
         case GT_NOT:
         case GT_NEG:
+            // Remove double negation
+            if (oper == GT_NEG && op1->OperIs(GT_NEG))
```

Comment:
Nit: It is strange to see an `if` on `switch` operand.
I would do 
```
case GT_NEG:
	if (op1->OperIs(GT_NEG))
	{
	    // Remove double negation
		GenTree* child = op1->AsOp()->gtGetOp1();
		return child;
	}
	__fallthrough;
case GT_NOT:
	old code.

```

Reply:
Sure thing, it wasn't clear to me what the preferred style is, to keep the commonality or separate cases.